src/components/NewsletterSignup/NewsletterSignup.mdx

```diff
@@ -0,0 +1,124 @@
+import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";
+
+import * as NewsletterSignupStories from "./NewsletterSignup.stories";
+import Link from "../Link/Link";
+
+<Meta of={NewsletterSignupStories} />
+
+# NewsletterSignup
+
+| Component Version | DS Version   |
+| ----------------- | ------------ |
+| Added             | `Prerelease` |
+| Latest            | `Prerelease` |
+
+## Table of Contents
+
+- {<Link href="#overview" target="_self">Overview</Link>}
+- {<Link href="#component-props" target="_self">Component Props</Link>}
+- {<Link href="#accessibility" target="_self">Accessibility</Link>}
+
+## Overview
+
+<Description of={NewsletterSignupStories} />
+
+**Note**: The `NewsletterSignup` example
+below alternately renders the "confirmation" and "error" screens on each form
+submission. This is just to demonstrate the different states of the component.
+In practice, the consuming app is responsible for handling the form submission.
+
+## Component Props
+
+<Canvas of={NewsletterSignupStories.WithControls} />
+
+<ArgTypes of={NewsletterSignupStories.WithControls} />
+
+## Accessibility
+
+The `NewsletterSignup` component is a complex component built from various Reservoir
+DS and Chakra components. The two main components are the DS `Button` component
+used to open Chakra's `Drawer` component.
+
+When the primary button is clicked, the dialog opens and focus is set to the
+first focusable element which is the "close" button that contains minus icon in
+the header of the dialog. While opened, focus is trapped within the dialog until
+it is closed either by clicking on the "close" or "Cancel" buttons, pressing the
+"Escape" key, or by clicking outside of the dialog. When the `NewsletterSignup`
+component is closed, focus is set back to the primary button that opened the
+dialog.
+
+The markup of the `NewsletterSignup` structurally matches the modal dialog pattern
+that is implemented by Chakra's `Modal` and `Drawer` components. The container
+has `role=”dialog”`, `aria-modal=”true”`, `tabindex={0}`, `aria-labelledby` that
+references the title within the dialog, and `aria-describedby` that references a
+descriptive piece of text within the dialog.
+
+Within the `NewsletterSignup` component, the radio input field is created from the DS
+`RadioGroup` and `Radio` components, and input fields are created from the DS
+`TextInput` component. Each of these components has their own accessibility
+features documented in their respective Storybook pages.
+
+When the form is submitted, focus is set to the confirmation message or the
+error message if an error occurs.
+
+Whereas the `NewsletterSignup`'s primary button element is placed within the DOM
+structure where it is rendered, the dialog DOM structure is appended to the end
+of the DOM tree and it is done by Chakra.
```

Comment:
All of this can be removed. But do include information regarding the focus on the confirmation and error views.